deployment/lib/identities.ts

```diff
@@ -45,6 +45,7 @@ export class Identities {
 
     props.buildBucket.grantRead(testRole, '*/dist/*');
     props.buildBucket.grantPut(testRole, '*/dist/*/tests/*');
+    props.buildBucket.grantPut(testRole, '*/test-results/*');
```

Comment:
Isn't this redundant with `*/dist/*/tests/*` we have above? When do we publish `test-results`?

Reply:
Its redundant but since we are in between the release not sure if this role is being used by anything else so just adding what is required right now to get test-publisher in. @owaiskazi19 upon you to check and remove the extra permissions or create an issue to modify those later.

Reply:
Either delete the duplicate or merge and open an issue please.

Reply:
Second option sounds good to me for now.

Reply:
Created https://github.com/opensearch-project/opensearch-build/issues/935 for the same.